### ISSUE TITLE
Crash when backgrounding the app after clearing user defaults.

### ISSUE THREAD

**Issue by gransell** [BranchMetrics/Branch-iOS-SDK#119]
Hi

We are running version 0.7.7 and found a crash bug. It happens when backgrounding the app after calling 
```
[[NSUserDefaults standardUserDefaults] removePersistentDomainForName:domainName];
```

```
Callstack is:
#0	0x00e7dad6 in -[Branch processNextQueueItem] at /Users/gransell/src/iphone-clean/Pods/Branch/Branch-SDK/Branch-SDK/Branch.m:1360
#1	0x00e7cd97 in -[Branch callClose] at /Users/gransell/src/iphone-clean/Pods/Branch/Branch-SDK/Branch-SDK/Branch.m:1253
#2	0x035206d9 in __NSFireTimer ()
#3	0x060ee866 in __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ ()
#4	0x060ee1ed in __CFRunLoopDoTimer ()
#5	0x060ac54a in __CFRunLoopRun ()
#6	0x060aba5b in CFRunLoopRunSpecific ()
#7	0x060ab88b in CFRunLoopRunInMode ()
#8	0x0767c2c9 in GSEventRunModal ()
#9	0x0767c106 in GSEventRun ()
#10	0x0424a106 in UIApplicationMain ()
#11	0x0007943a in main at /Users/gransell/src/iphone-clean/qapital/Supporting Files/main.m:16
#12	0x06a3fac9 in start ()
```

Let me know if you need any thing else.

**Comment by shortstuffsushi** [comment]
Hi @gransell. Thanks for the report, but unfortunately it's not immediately obvious to me what the failure is here. We do store some items in NSUserDefaults, but I'm not really familiar with the domains concept. Which domain is it that you're clearing? I'll keep looking to see if I can reproduce.

**Comment by shortstuffsushi** [comment]
Also, what is the actual crash on? The error you've got there is the stack trace, but doesn't mention what the error is. Could you add that?

**Comment by shortstuffsushi** [comment]
Never mind, I found this issue, and fixed it in f22d05c1. I will be releasing an update shortly, sorry about that!

**Comment by gransell** [comment]
No problems. Thanks for the update.
